flytekit/clients/friendly.py

```diff
@@ -353,12 +370,19 @@ def create_launch_plan(
             the identical launch plan is already registered.
         :raises grpc.RpcError:
         """
-        super(SynchronousFlyteClient, self).create_launch_plan(
-            _launch_plan_pb2.LaunchPlanCreateRequest(
-                id=launch_plan_identifer.to_flyte_idl(),
-                spec=launch_plan_spec.to_flyte_idl()
-            )
+        create_request = _launch_plan_pb2.LaunchPlanCreateRequest(
+            id=launch_plan_identifer.to_flyte_idl(),
+            spec=launch_plan_spec.to_flyte_idl()
         )
+        try:
+            super(SynchronousFlyteClient, self).create_launch_plan(create_request)
+        except _grpc.RpcError as e:
+            # When Admin detects a conflicting entry or otherwise invalid request during registration, it returns an
+            # invalid argument code. We should print out the thing we were trying to create in these cases.
+            if e.code() == _grpc.StatusCode.INVALID_ARGUMENT:
```

Comment:
why not dump all other error codes here? This way we could probably make the error message better?